tests/61push/08_rejected_pushers.pl

```diff
@@ -20,17 +23,22 @@ sub create_pusher
 
 sub wait_for_push
 {
-   my ( $path, $response ) = @_;
+   my ( $pushkey, $response ) = @_;
 
-   await_http_request( $path, sub {
+   await_http_request( $PUSH_LOCATION, sub {
       my ( $request ) = @_;
       my $body = $request->body_from_json;
 
-      # Respond to all requests, even if we filter them out
-      $request->respond_json( $response // {} );
-
       return unless $body->{notification}{type};
       return unless $body->{notification}{type} eq "m.room.message";
+
+      # Ensure this is the expected pusher.
+      return unless $body->{notification}{devices};
+      return unless $body->{notification}{devices}[0]{pushkey} eq $pushkey;
+
+      # Respond to expected request.
+      $request->respond_json( $response // {} );
+
       return 1;
    });
```

Comment:
This is attempting to check one of the fields of the request instead of differentiating on URL.